### ISSUE TITLE
Feature: Endpoint non-response/failure are included in average response time

### ISSUE THREAD

**Issue by michaelkrieger** [TwiN/gatus#426]
### Describe the feature request

Incorrectly, **the response time graph (and the averages for 1hr/24hr/7d) are influenced by unavailability.**
Correctly, the uptime percentages seems to correctly reflect how often a service is unavailable.

**If a ping is always 5-7ms but fails 1% of the time, the resulting average response time is 175ms and the graph shows that high figure.** (for example)

An option should be added (or it should probably be default) to:
1. Remove the outlier non-response from the response times. This data is already captured in the 'uptime percentage' figures and it is not a measure of response time at all.
2. If desired, add a bar graph (in addition to the existing line graph) as a second Y axis on the existing line graph showing 'percent unavailability'. This is a very common graph type.  It would be appropriate for it to share the same graph and just have a bar graph on the line graph.

### Why do you personally want this feature to be implemented?

Having any sort of packet loss or downtime results in huge inaccurate spikes in response time. This makes the graph and the averages totally inaccurate.  They lose meaning when a 5ms ping time includes things that are not a ping time at all.

### How long have you been using this project?

5 weeks

**Comment by TwiN** [comment]
I understand where you're coming from, but let's say that you have an endpoint configured with the following condition:
```
  - "[RESPONSE_TIME] < 500"
```
If we went with the approach you're suggesting, any time the condition would fail (i.e. the response time would be above 500ms), we wouldn't exclude it in the average response time, and I feel like that wouldn't be representative of what a user would expect when they see "average response time".

**Comment by michaelkrieger** [comment]
> any time the condition would fail (i.e. the response time would be above 500ms), we wouldn't exclude it in the average response time, and I feel like that wouldn't be representative of what a user would expect when they see "average response time".

But what value -is- used in that case? If a test has a condition of <500ms, is Gatus waiting some arbitrary time like ?2000ms? anyway so it can plot the actual number? Is it cancelling the request at 500ms when it knows the condition won’t be met? From the docs I can’t see any timeout being set so I assume there’s a default ICMP and HTTP max in there. 

**there is a very big difference between no response and a slow response that exceeds threshold.**

Looking at a simple ping with maybe a timeout of 500ms can:
 - be within the required time and thus be the time
 - Be a figure which exceeds 500ms but which Gatus waits for a response anyway and thus has an accurate number
 - be set at 500ms which is the max we are testing, thus allowing us to move on to the next test.
 - Log as a failure and do not include in calculations
 
Maybe the answer is a configurable outcome?  on_failure or on_noresponse with options like “ignore”, “exclude_actual”, or “log_500”?

I can see uses in both cases now that you point it out, but I can tell you that knowing my ping to my ISP is 170ms because I rebooted the router somewhere in the middle of 5ms pings makes this a meaningless figure and prevents seeing trends and having a true measure of speed.

as an analogy, I’m sure a 50mph road with cars travelling 50mph isn’t considered a 30mph road because one car is pulled over at the side of the road changing a flat tire.

**Comment by MarioLiebisch** [comment]
I agree with Michael here. But let's spin it to a more generic use case:

Let's say my average response time is 5-15ms and uptime is 100%. That's excellent.

Now I perform maintenance, the server stays down for an hour. This will result in timeouts for an hour, too.

The uptime for the last hour will be 0%, uptime for the last 24 hours will be like 95%, etc. that's totally fine, but at the same time – based on my observations – the average response time will no longer be 5-15ms, but significantly higher due to all those 10 second timeouts factored in. However, this won't represent the user's average experience, not at all.

To be fair, just recently started using Gatus and it's totally possible I'm misinterpreting something here, but Michael's point is something I also noticed immediately while still configuring everything and I agree, this feels totally off.

**Comment by srgykuz** [comment]
Up. 

In monitoring systems that I worked before with it was a common (or, at least, optional) to exclude error results from the final metrics that are related to time in order to not distort them. Let's say a website that Gatus is monitoring was unavailable for an hour and failed with 10s timeout. Now all the response time metrics are distorted by that. Normally response time of the website is ~100ms, but now I see: "average response time - 4500ms", "response time ranges - 76-10000ms", "response time 7d - 2988ms" - that's all doesn't make sense anymore. The metrics became useless; they don't reflect anything anymore.

It would be a nice feature to exclude error results from all response time metrics. Other metrics, like Uptime Statistics, should continue including error results.
